src/large_pages/node_large_page.cc

```diff
@@ -258,23 +258,21 @@ struct text_region FindNodeTextRegion() {
 
 #if defined(__linux__)
 bool IsTransparentHugePagesEnabled() {
-  std::ifstream ifs;
-
   // File format reference:
   // https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/mm/huge_memory.c?id=13391c60da3308ed9980de0168f74cce6c62ac1d#n163
-  ifs.open("/sys/kernel/mm/transparent_hugepage/enabled");
-  if (!ifs) {
+  const char* filename = "/sys/kernel/mm/transparent_hugepage/enabled";
+  std::ifstream config_stream(filename, std::ios::in);
+  if (!config_stream.good()) {
     PrintWarning("could not open /sys/kernel/mm/transparent_hugepage/enabled");
```

Comment:
Note for review: this changeset does introduce a small behaviour change; the `PrintWarning` function will now be called if the config file fails to open, whereas no `else` condition is present in the version at line https://github.com/nodejs/node/pull/37187/files#diff-930e2e7e309763acbb831b1cc081a2a76c92e2a3d0be150a9e933bdc7a1747f0L272